src/lib/components/SearchResults/index.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * @fileoverview An Algolia search box.
+ */
+
+import {html} from 'lit-element';
+import {unsafeHTML} from 'lit-html/directives/unsafe-html';
+import {BaseElement} from '../BaseElement';
+import {allowHtml, escapeHtml} from '../../../lib/utils/escape-html';
+import 'focus-visible';
+import './_styles.scss';
+
+/**
+ * An Algolia search box.
+ * @extends {BaseElement}
+ * @final
+ */
+class SearchResults extends BaseElement {
+  static get properties() {
+    return {
+      // An array of algolia results.
+      hits: {type: Object},
+      // An internal array of algolia results (observable).
+      hits_: {type: Object},
+      // Manages showing/hiding the search results element.
+      showHits: {type: Boolean},
+      // Indicates which search result should be highlighted.
+      // Primarily used for keyboard behavior.
+      cursor: {type: Number},
+      // Search query
+      query: {type: String},
+    };
+  }
+
+  constructor() {
+    super();
+    this.hits = [];
+    this.hits_ = [];
+    this.showHits = false;
+    this.cursor = -1;
+    this.query = '';
+  }
+
+  set hits(hits) {
+    this.hits_ = hits;
+    this.cursor = -1;
+  }
+
+  navigate(key) {
```

Comment:
Can you add a comment explaining why this method exists and how it's used by another element to essentially drive this element.